modules/vba_documents/app/sidekiq/vba_documents/upload_processor.rb

```diff
@@ -60,14 +66,11 @@ def download_and_process
         validate_metadata(parts[META_PART_NAME], @upload.consumer_id, @upload.guid,
                           submission_version: @upload.metadata['version'].to_i)
         metadata = perfect_metadata(@upload, parts, timestamp)
+        metadata['ICN'] = icn if icn.present?
```

Comment:
Pass ICN along to EMMS in the metadata we send them as part of out upload request.

Reply:
Now that `icn` is stored in the metadata on the `@upload`, which is passed as an argument to `perfect_metadata`, I think that the assigning step should move to the `perfect_metadata` method so that it's done alongside all other field assignments. (`UploadProcessor` is large enough as it is.)

Also, looking at `perfect_metadata`, I noticed that every field other than `ICN` is camel case. Did EMMS just not follow their naming convention with that field, or is it possible that we're not passing the right key?

Reply:
Good idea, thanks.

I'll save our stored version as icn

I just re-confirmed, they did tell me and give me a copy of their Interface Control Document For DOMA EMMS Benefits Intake API, and yep, they want it all caps: "ICN"....